src/Features/LanguageServer/Protocol/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs

```diff
@@ -123,17 +123,14 @@ async Task AddDocumentsAndProject(Project? project, ImmutableArray<string> suppo
                     return;
                 }
 
-                var isFSAOn = globalOptions.IsFullSolutionAnalysisEnabled(project.Language);
-                var documents = ImmutableArray<TextDocument>.Empty;
-                // If FSA is on, then add all the documents in the project.  Other analysis scopes are handled by the document pull handler.
-                if (isFSAOn)
-                {
-                    documents = documents.AddRange(project.Documents).AddRange(project.AdditionalDocuments);
-                }
+                // Only process closed files if FSA is on.
+                if (!globalOptions.IsFullSolutionAnalysisEnabled(project.Language))
+                    return;
```

Comment:
this logic appears to be what we all want.  if FSA is not on, then we don't process closed documents.